gtamsolution.txt

```diff
@@ -0,0 +1,233 @@
+
+console.log('Exercise 4 Result:\n', favFood);
+
+/*
+Exercise 5:
+  - Insert the string 'tofu' in the foods array between 'pizza' & 'cheeseburger'
+*/
+
+// Complete Exercise 5 below...
+// .splice: arguments start position, number of elements to delete, elements to add
+foods.splice(2,0,'tofu')
+
+console.log('Exercise 5 Result:\n', foods);
+
+/*
+Exercise 6:
+  - Replace the string 'pizza' in the foods array with the strings 'sushi' & 'cupcake'.
+*/
+
+// Complete Exercise 6 below...
+//console.log(foods.indexOf('pizza')) //finds location of pizza
+foods.splice(1, 1, 'sushi', 'cupcake')
+
+console.log('Exercise 6 Result:\n', foods);
+
+/*
+Exercise 7:
+  - Use the slice method on the foods array to create a new array containing 'sushi' & 'cupcake'.
+  - Assign the new array to a variable named yummy.
+*/
+
+// Complete Exercise 7 below...
+var yummy= foods.slice(1,3);
+
+console.log('Exercise 7 Result:\n', yummy);
+
+/*
+Exercise 8:
+  - Using the indexOf method on the foods array, assign the index of the 'tofu' string to a variable named soyIdx.
+*/
+
+// Complete Exercise 8 below...
+var soyIdx= foods.indexOf('tofu')
+
+console.log('Exercise 8 Result:\n', soyIdx);
+
+/*
+Exercise 9:
+  - Assign to a variable named allFoods the result of joining the strings in the foods array such that the result is the following single string:
+    'taco -> sushi -> cupcake -> tofu -> cheeseburger'
+*/
+
+// Complete Exercise 9 below...
+var allFoods= foods.join(' -> ')
+
+console.log('Exercise 9 Result:\n', allFoods);
+
+/*
+Exercise 10:
+  - Assign a boolean to a variable named hasSoup depending upon whether or not the foods array includes the string 'soup'.
+*/
+
+// Complete Exercise 10 below...
+var hasSoup= foods.includes('soup') ;
+
+console.log('Exercise 10 Result:\n', hasSoup);
+
+
+/*
+Exercise 11:
+  - Use the forEach method to iterate through the provided nums array and add each odd number to a new array named odds.
+  - Hint: Initialize the odds variable to an empty array before the iteration.
+*/
+
+const nums = [100, 5, 23, 15, 21, 72, 9, 45, 66, 7, 81, 90];
+
+// Complete Exercise 11 below...
+ var odds= [];
+ nums.forEach(function(item, index){
+   if(item%2==1){
+     odds.push(item);
+   }
+
+ });
+
+console.log('Exercise 11 Result:\n', odds);
+
+/*
+Exercise 12:
+  - Use the forEach method to iterate through the same nums array and add the number to arrays named fizz, buzz and/or fizzbuzz based upon the following:
+  	- Add to the fizz array if the number is evenly divisible by 3.
+  	- Add to the buzz array if the number is evenly divisible by 5.
+  	- Add to the fizzbuzz array if the number is evenly divisible by 3 & 5.
+*/
+
+// Complete Exercise 12 below...
+var fizz= [];
+var buzz= [];
+var fizzbuzz= [];
+
+nums.forEach(addTo);
+
+function addTo(item){
+  if(item % 3 == 0 && item % 5 ==0){
+    fizzbuzz.push(item);
+    if(item% 3==0) {
+      fizz.push(item);
+      if(item%5 ==0){
+        buzz.push(item);
+      }
+    }
```

Comment:
interesting! you're already doing a check to see if the number is divisible by 3 and 5 in line 143, why are you doing additional checks for those same conditions in lines 145 and 147? You could do something like:

```
if (item % 3 === 0 && item % 5 === 0){
    fizzbuzz.push(item)
    fizz.push(item)
    buzz.push(item)
}
```